### DISCUSSION TITLE
Error message when I run "serve" command: Uncaught SyntaxError: expected expression, got ')'

### DISCUSSION THREAD

**Discussion by agustin-bilat** [GMOD/jbrowse-components#3657] [Q&A] [answered]
After following the [steps](https://jbrowse.org/jb2/docs/quickstart_web/) in order to install JBrowse Web I finally run the `npx serve .` command inside de jbrowse2/ directory (obtained with `jbrowse create jbrowse2`) and when I opened the local port  `http://localhost:3000` I get an empty white page. 

By checking the Web developer tools's debugger I get the error message mentioned in the title (I am using Mozilla Firefox 112.0.1) .

It seems that there is a ')' somewhere but I have no idea how to find it nor how to fix it. When I click, "view source in debugger" though, I notice that the very first character is a ')'  : 

> ){case 107:return u(e,":",":"+F)+e;case 101:return u(e,/(.+:)([^;!]+)(;|!.+)?/,"$1"+F+(45===s(e,14)?"inline-":"")+"b

....etcetera.

`jbrowse --version`

> @jbrowse/cli/2.4.2 linux-x64 node-v18.16.0

`npx serve --version`
> 14.2.0

Any help would be much appreciated

Best,
Agustin.

**Comment by garrettjstevens** [comment]
I've tried this with the same version of `jbrowse`, `serve`, and Firefox and I can't reproduce the problem. Have you tried a different browser? Also, if you could copy the full error message in the console, including any file name, that could help diagnose it. If you could copy the output of the `npx serve .` command, too, that might help.

**Reply by agustin-bilat** [reply]
> Have you tried a different browser?

I did, I use google chrome too with the same result.

The image displayed in the console is this:
<img width="344" alt="npx" src="https://user-images.githubusercontent.com/82325348/233706236-b841b6d9-5b7d-4dc6-91af-329212124221.png">

And the message within the console after navigating into the port `http://localhost:3000`  as it is seen in the Web dev. tools is:

> Uncaught SyntaxError: expected expression, got ')'   [Learn More]           main.28981932.js:1

If I click main.28981832.js:1 I get this file
[Untitled.js.txt](https://github.com/GMOD/jbrowse-components/files/11298363/Untitled.js.txt)

Please let me know if you need any further information.

**Reply by garrettjstevens** [reply]
That `main.28981832.js` is missing the beginning, I'm not sure why that would happen. Does the same thing happen if you completely remove the `jbrowse2/` directory and create it again with `jbrowse create jbrowse2`?

**Comment by agustin-bilat** [comment]
You're right, I've just made what you mentioned (removing the  `jbrowse2/` and create it again) and what I get as a result is a different syntax error:

> Uncaught SyntaxError: expected expression, got '}'

The corresponding  file is now:
[main.28981932.js:1:8](https://github.com/GMOD/jbrowse-components/files/11298606/Untitled.js.txt)

**Reply by garrettjstevens** [reply]
It looks like that file is missing the beginning as well. You could try installing it manually. To do this, first you need to remove the `jbrowse2` directory and then run these commands:
```sh
curl -Lo jbrowse.zip https://github.com/GMOD/jbrowse-components/releases/download/v2.4.2/jbrowse-web-v2.4.2.zip
unzip -d jbrowse2 jbrowse.zip
rm jbrowse.zip
```
Let me know if this works for you.

**Reply by agustin-bilat** [reply]
Thanks, I did what you've just mentioned and at least I can see the Volvox sample config file now (not the others though).
![image](https://user-images.githubusercontent.com/82325348/233739559-87591a73-916a-4eb6-827b-470a8ad4dfad.png)

**Reply by garrettjstevens** [reply]
The screenshot in the tutorial is out of date, only seeing Volvox is what is expected, so I think you have everything working now.

**Comment by agustin-bilat** [comment]
Many thanks for your help and your patience, it seems to be working now.

Best Regards,
Agustin.

**Comment by nathanweeks** [comment]
I've noticed a similar error recently when jbrowse is installed with `jbrowse create`. It seems `jbrowse create`  results in a different main.d42b84c7.js vs. manually downloading/unzipping jbrowse-web-v2.5.0.zip:
```sh
$ curl -Lo jbrowse.zip https://github.com/GMOD/jbrowse-components/releases/download/v2.5.0/jbrowse-web-v2.5.0.zip
...
$ unzip jbrowse.zip
...
$ echo "$(head -c 50 static/js/main.d42b84c7.js)"
/*! For license information please see main.d42b84
$ jbrowse --version
@jbrowse/cli/2.5.0 linux-x64 node-v18.16.0
$ jbrowse create broken
Fetching https://github.com/GMOD/jbrowse-components/releases/download/v2.5.0/jbrowse-web-v2.5.0.zip...
Unpacked https://github.com/GMOD/jbrowse-components/releases/download/v2.5.0/jbrowse-web-v2.5.0.zip at broken
$ echo "$(head -c 50 broken/static/js/main.d42b84c7.js)"
41===e&&L(e);break;case 92:E()}return y}function R
```
```

**Reply by nathanweeks** [reply]
The error is not observed with node 16.x:
```sh
$ nvm use 16
Now using node v16.20.0 (npm v8.19.4)
$ jbrowse --version
@jbrowse/cli/2.5.0 linux-x64 node-v16.20.0
$ jbrowse create not-broken
Fetching https://github.com/GMOD/jbrowse-components/releases/download/v2.5.0/jbrowse-web-v2.5.0.zip...
Unpacked https://github.com/GMOD/jbrowse-components/releases/download/v2.5.0/jbrowse-web-v2.5.0.zip at not-broken
$ echo "$(head -c 50 not-broken/static/js/main.d42b84c7.js)"
/*! For license information please see main.d42b84
```

**Reply by cmdcolin** [reply, marked as answer]
@nathanweeks thanks for the extra info. I think the fact it occurs at newer versions of node is true, the nodejs library we used (node-unzipper) has some issues on their github about this. I changed the libraries in this PR https://github.com/GMOD/jbrowse-components/pull/3686 and hopefully can get it released soon!

**Reply by cmdcolin** [reply]
can see a bunch here https://github.com/ZJONSSON/node-unzipper/issues/271 https://github.com/ZJONSSON/node-unzipper/issues/278 maybe https://github.com/ZJONSSON/node-unzipper/issues/269

**Reply by nathanweeks** [reply]
@cmdcolin Thanks! It's good to know a fix is slated for the next release, and that there's an easy workaround in the meantime.
